### ISSUE TITLE
using `github.com/bufbuild/buf/cmd/buf` with go generate

### ISSUE THREAD

**Issue by frantjc** [bufbuild/buf#1233]
Hello!

I am trying to set up my project so that go generate uses `buf` like so:

```go
package generate

//go:generate go run -tags generate github.com/bufbuild/buf/cmd/buf generate .
```

This is preferable because then the version of `buf` being used by the project will be consistent across machines and it eliminates one step of setting up dependencies.

However, go generate requires the dependency be in `go.mod`:

```
require (
	github.com/bufbuild/buf v1.6.0 // indirect
)
```

Alas, go mod marks this as an unused dependency. Normally you'd use a dummy import to get around this:

```
package tools

import _ "github.com/bufbuild/buf/some/pkg"
```

The problem with this approach regarding `github.com/bufbuild/buf` is that it has no importable packages!

So, in summary, my request is for any importable go package to be added to the `github.com/bufbuild/buf` module so that it can be used as a dummy import to make go mod not mark the module as unused

**Comment by frantjc** [comment]
whoops, found some packages under `private` that will hold me over. Might leave this open as it feels a bit wrong to use a package under `private` as a dummy import--I'll let the maintainers decide, I suppose.

**Comment by frantjc** [comment]
any in case anyone comes across this same situation later: I ran into some problems with importing most of the packages under `private` until I found `github.com/bufbuild/buf/private/buf/cmd/buf`

**Comment by frantjc** [comment]
new development:

```sh
panic: github.com/bufbuild/buf/private code must only be imported by github.com/bufbuild projects
```

so I am back to square one with my request :)

**Comment by bufdev** [comment]
We wouldn't recommend the blank import route for `buf` or any other tool, there's issues with this paradigm, notably that it means tools get built with different versions of libraries that overlap with your own code than what they were intended to be built with. Alas, we know people do use this paradigm, but regardless, we don't want to expose any of `buf` as a library - given our strict adherence to SemVer, this would mean not being able to break the internal implementation of this repository, which we do all the time. We want to limit the surface area of `buf` to be the command itself. We use `private` instead of `internal` as we share this code with our internal repositories, and you found out how we limit usage outside of github.com//bufbuild, which we aren't going to change. So I think unfortunately we can't fulfill this request, our apologies.
